editoast/src/views/timetable/similar_schedules.rs

```diff
@@ -0,0 +1,124 @@
+use axum::Extension;
+use axum::Json;
+use axum::extract::State;
+use chrono::DateTime;
+use chrono::Utc;
+use editoast_authz::Role;
+use serde::Deserialize;
+use serde::Serialize;
+use utoipa::ToSchema;
+
+use crate::error::Result;
+
+use super::AppState;
+use super::AuthenticationExt;
+use super::AuthorizationError;
+
+editoast_common::schemas! {
+    Waypoint,
+    RollingStockCharacteristics,
+    SimilarSchedulesRequest,
+    SimilarScheduleResponse,
+}
+
+crate::routes! {
+    "/similar_schedules" => {
+        similar_schedules,
+    },
+}
+
+#[derive(Debug, Deserialize, ToSchema)]
+#[allow(dead_code)]
+struct RollingStockCharacteristics {
+    name: String,
+    towed_rolling_stock: Option<String>,
+    speed_limit_tag: Option<String>,
+    mass: Option<u64>,
+    // #[serde(with = "editoast_common::units::kilogram::u64")]
+    // mass: Option<uom::si::u64::Mass>,
+}
+
+#[derive(Clone, Serialize, Deserialize, ToSchema)]
+#[cfg_attr(test, derive(PartialEq))]
+#[schema(as = SimilarScheduleWaypoint)]
+struct Waypoint {
+    ci: i64,
+    ch: String,
+    stop: bool,
+}
+
+impl std::fmt::Debug for Waypoint {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "{}:{}{}",
+            self.ci,
+            self.ch,
+            if self.stop { "[STOP]" } else { "" },
+        )
+    }
+}
+
+#[derive(Debug, Deserialize, ToSchema)]
+#[allow(dead_code)]
+struct SimilarSchedulesRequest {
+    #[schema(inline)]
+    rolling_stock: RollingStockCharacteristics,
+    #[schema(inline)]
+    waypoints: Vec<Waypoint>,
+}
+
+#[derive(Debug, Serialize, ToSchema)]
+struct SimilarScheduleResponse {
```

Comment:
```suggestion
struct SimilarScheduleResponse {
```

ill named : component of a response not the whole object

Reply:
Done.